docs/contract-test-collection.md

```diff
@@ -0,0 +1,526 @@
+
+That may not sound very useful, but the brilliant part is that the error we receive contains the information we requested! We can not incur a fee for requesting data or waiting for confirmation from the network to call view methods.
+
+Taquito internally contains a list of lambda contracts. Thus, there is no need to deploy a lambda contract if you are using Mainnet, Hangzhounet, or Granadanet. Taquito will detect the current network and use the appropriate lambda contract.
+
+Lambda views are introduced in [Tzip4](https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-4/tzip-4.md#view-entrypoints).
+
+## LambdaViewContract
+
+[KT1LNMrk8orMQ85zbwK25996dPhDxfSicvKh](https://better-call.dev/hangzhou2net/KT1LNMrk8orMQ85zbwK25996dPhDxfSicvKh/code)
+
+Not a supported FA1.2 contract. Almost an Fa2 interface but it is missing update_operators.
+
+#### Entrypoints:
+  * approve
+  * getAllowance
+  * getBalance
+  * getTotalSupply
+  * mint
+  * transfer
+
+#### Storage:
+```js
+storage (pair
+          (pair
+            (big_map %ledger address (pair (map %allowances address nat) (nat %balance)))
+            (address %owner))
+          (nat %totalSupply));
+```
+
+## LambdaViewWithTokenContract
+
+[KT1Tf2JXZP8wXjdgndsgMKM1uW9M4CC5rbWL](https://better-call.dev/hangzhou2net/KT1Tf2JXZP8wXjdgndsgMKM1uW9M4CC5rbWL/code)
+
+This contact is another example of a Lambda contract, this time involving a token. It is not a supported FA1.2 contract but does have an FA2 interface.
+
+#### Entrypoints:
+  * balance_of
+  * token_metadata_registry
+  * transfer
+  * update_operators
+
+#### Storage:
+```js
+storage (pair
+          (pair
+            (big_map %ledger address (pair (set %allowances address) (nat %balance)))
+            (big_map %token_metadata nat
+                                     (pair (nat %token_id)
+                                           (pair (string %symbol)
+                                                 (pair (string %name)
+                                                       (pair (nat %decimals)
+                                                             (map %extras string string)))))))
+          (nat %total_supply));
+```
+
+# Contracts with Maps
+
+## MapWithPairasMapContract
+
+[KT1BhFn1n1h4HJCxaRvoWEHPcp5UpAYbH3XN](https://better-call.dev/hangzhou2net/KT1BhFn1n1h4HJCxaRvoWEHPcp5UpAYbH3XN/code)
+
+A simple contract with a default entrypoint that takes unit. Not a supported FA1.2 contract. 
+
+#### Entrypoints:
+  * default
+#### Storage:
+```js
+storage (pair
+          (pair (address %theAddress)
+                (map %theMap (pair nat address) (pair (mutez %amount) (int %quantity))))
+          (int %theNumber));
+```
+Note the lack of annotations. If the storage does not annotate its properties, the caller must use numeric indexes instead.
```

Comment:
Doesn't really apply to the contract MapWithPairasMapContract because it has annotations

Reply:
Fixed. Moved it to MapWithComplexKeysContract, which has no annotations.

In the next version, the Taquito code for setting the storage will be included, and this one will be a good example.